### PR TITLE
Offline Mode: Instrumentation

### PR DESCRIPTION
**Pull request by kean**
This PR adds some basic instrumentation:

- Add `WPAssert` to capture assertions in production
- Add tracking for post sync errors (also using Tracks)

To test: n/a

## Regression Notes
1. Potential unintended areas of impact: n/a
2. What I did to test those areas of impact (or what existing automated tests I relied on): n/a
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/Utility/WPInstrumentation.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import Foundation
+import Sentry
```
Comment (momo-ozawa) [inline WordPress/Classes/Utility/WPInstrumentation.swift:2]:
```suggestion
```

File: WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift [hunk 2]
```diff
@@ -1541,6 +1545,12 @@ import Foundation
         case .widgetsLoadedOnApplicationOpened:
             return "widgets_loaded_on_application_opened"
 
+        // Assertions & Errors
+        case .assertionFailure:
+            return "assertion_failure"
+        case .postCoordinatorErrorEncountered:
+            return "post-coordinator-eerror-encountered"
```
Comment (momo-ozawa) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift:1563]:
```suggestion
            return "post_coordinator_error_encountered"
```

File: WordPress/Classes/Services/PostCoordinator.swift [hunk 3]
```diff
@@ -261,6 +266,15 @@ class PostCoordinator: NSObject {
         topViewController.present(alert, animated: true)
     }
 
+    private func trackError(_ error: Error, operation: String) {
+        DDLogError("post-coordinator-\(operation)-failed: \(error)")
+
+        if let error = error as? TrackableErrorProtocol, var userInfo = error.getTrackingUserInfo() {
+            userInfo["operation"] = operation
+            WPAnalytics.track(.postCoordinatorErrorEncountered, properties: userInfo)
```
Comment (momo-ozawa) [inline WordPress/Classes/Services/PostCoordinator.swift:273]: Just wanted to note that for Tracks, snake case is preferred. But this is a property value so I don't have strong feelings - I'll let you decide!

**Review by momo-ozawa — COMMENTED** [review]
Works as described! Can you register/validate the events? Thank you!

Screenshot 2024-04-10 at 17 01 38

File: WordPress/Classes/Services/PostCoordinator.swift (hunk 3 above)
Comment (kean) [reply WordPress/Classes/Services/PostCoordinator.swift:273]: Oh, for property names. Totally – updated.

**Review by momo-ozawa — APPROVED** [review]

**Pull request merged**